### DISCUSSION TITLE
Question regarding opacity utils

### DISCUSSION THREAD

**Discussion by eirikove** [tailwindlabs/tailwindcss#9226] [Help]
Hi Tailwind, thanks for the great library!

We recently changed from `prettier-plugin-tailwindcss` to `eslint-plugin-tailwindcss` which gives us a few errors on the [opacity syntax](https://tailwindcss.com/docs/upgrade-guide#new-opacity-modifier-syntax)

We're finding this slightly annoying on standardized components with opacity as a [vue fallthrough attribute](https://vuejs.org/guide/components/attrs.html); an example of our code is something like this

```vue
<!-- Box.vue -->
<template>
  <div class="bg-white ...">
    <slot />
  </div>
</template>

<!-- usage -->
<Box class="bg-opacity-50"> <!-- lint error here -->

<!-- not so great workaround 1 -->
<!-- eslint-disable tailwindcss/migration-from-tailwind-2 -->
  <Box class="bg-opacity-50">
<!-- eslint-enable --> 

<!-- not so great workaround 2 -->
<Box class="bg-white/50">
```

Your migration docs are saying opacity utilities will be disabled come v4
 
>  We do recommend using the new syntax going forward though, and plan to disable utilities like border-opacity-* and bg-opacity-* by default in v4, though you will still be able to enable them if needed.

I do enjoy this new syntax, but I'm curious how we can (and how you're planning to) solve for fallthrough classes without using the current opacity utilities. Is there a good argument for doing duplicate markup for coloring? (i.e. the not so great workaround 2)

**Comment by wongjn** [comment]
For now, you could use a CSS variable for this:

```html
<template>
  <div class="bg-white/[var(--opacity,1)] ...">
    <slot />
  </div>
</template>
```
```html
<Box class="[--opacity:0.5]">
```
